server/fleet/app.go

```diff
@@ -509,6 +505,32 @@ func (e *EnrichedAppConfig) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
+// MarshalJSON implements the json.Marshaler interface to make sure we serialize
+// both AppConfig and enrichedAppConfigFields properly:
+//
+// - If this function is not defined, AppConfig.MarshalJSON gets promoted and
+// will be called instead.
+// - If we try to unmarshal everything in one go, AppConfig.MarshalJSON doesn't get
+// called.
+func (e *EnrichedAppConfig) MarshalJSON() ([]byte, error) {
+	// Marshal only the enriched fields
+	enrichedData, err := json.Marshal(e.enrichedAppConfigFields)
+	if err != nil {
+		return nil, err
+	}
+
+	// Marshal the base AppConfig
+	appConfigData, err := json.Marshal(e.AppConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// combine the two byte arrays
```

Comment:
I'd add just a bit of context here, basically that both values marshal to JSON objects, so we're replacing the closing `}` of the first with a comma, and removing the starting `{` of the second. We should be careful never to have duplicate keys in both, though. I wonder if the code should make some checks before that combination, maybe including checks that first and last bytes of each are indeed `{` and `}` and fail otherwise, just to be super cautious? Just trying to make this tricky manipulation as robust as possible.

Reply:
that sounds good, I'll add some checks. Is there a better way to handle this scenario? I was hoping you would tell me that this is a really dumb way to go about this problem 😭

Reply:
I tried the map approach, the problem is that datetime fields were getting casted to `float64` :(

I pushed an initial approach for this at https://github.com/fleetdm/fleet/pull/13999/commits/82fdf611efaad4c8f91fc7339fc137d2d7bc4b3e , what do you think?

Reply:
@roperzh I think it looks very good with this `rawjson` package you created and the checks made there! I think that's fine like that, at some point we'll have a new major version and will be able to clean-up those configs a bit but for now we need this backwards compatibility support so this complexity is unavoidable.